### PR TITLE
feat: support plugin-basic-ssl

### PR DESCRIPTION
**Pull request by nanianlisao**
## Related Links

#2175 

## Checklist

- [ ] Tests updated (or not required).
- [x] Documentation updated (or not required).

### CODE REVIEW

File: packages/plugin-basic-ssl/package.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
+{
+  "name": "@rsbuild/plugin-basic-ssl",
+  "version": "0.6.5",
+  "description": "Ssl plugin for Rsbuild",
```
Comment (chenjiahan) [inline packages/plugin-basic-ssl/package.json:4]: SSL can be better 😄

```suggestion
  "description": "SSL plugin for Rsbuild",
```

File: website/docs/en/config/server/https.mdx [hunk 2]
```diff
@@ -23,17 +23,18 @@ HTTPS:
...
+import { pluginBasicSsl } from '@rsbuild/plugin-basic-ssl';
 
 export default {
+  plugins: [pluginBasicSsl()],
   server: {
-    https: {
-      key: fs.readFileSync('certificates/private.pem'),
-      cert: fs.readFileSync('certificates/public.pem'),
-    },
+    https: {},
```
Comment (chenjiahan) [inline website/docs/en/config/server/https.mdx:37]: Maybe we do not need users to set `server.https`.

Just register the plugin and HTTPS will work.

Comment (nanianlisao) [reply website/docs/en/config/server/https.mdx:37]: Nice idea.

**Review by chenjiahan — APPROVED** [review]

**Comment by chenjiahan** [comment]
Thanks! This is a good start for SSL support, and we can continue to improve this plugin in the future.

**Pull request merged**